<?xml version="1.0"?>
<launch>

    <!-- Load robot model into Parameter Server -->
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find MYROBOT_description)/urdf/MYROBOT.urdf.xacro'" />

    <!-- Start Robot State Publisher (calculates joint frame transforms) -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <!-- Start Joint State Publisher with joint control GUI -->
    <param name="use_gui" value="true" />
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"/>

    <!-- Start Rviz visualization tool -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find MYROBOT_description)/rviz/MYROBOT-fuerte.vcg" />

    <include file="$(find planning_environment)/launch/environment_server.launch">
        <arg name="use_monitor" value="true" />
        <arg name="use_collision_map" value="false" />
    </include>

    <rosparam command="load" ns="robot_description_planning" file="$(find MYROBOT_arm_navigation)/config/MYROBOT_planning_description.yaml" />

    <node pkg="arm_kinematics_constraint_aware" type="arm_kinematics_constraint_aware" name="MYROBOT_manipulator_kinematics" output="screen" >
        <param name="group" type="string" value="manipulator"/>
        <param name="manipulator/root_name" type="string" value="Mounting_link"/>
        <param name="manipulator/tip_name" type="string" value="Tool_link"/>

        <param name="kinematics_solver" type="string" value="arm_kinematics_constraint_aware/KDLArmKinematicsPlugin" />
        <!-- param name="kinematics_solver" type="string" value="MYROBOT_manipulator_kinematics/IKFastKinematicsPlugin" -->

   </node>

</launch> 

